craft_application/models/spread.py

```diff
@@ -28,7 +28,7 @@ class SpreadBase(CraftBaseModel):
 
     model_config = pydantic.ConfigDict(
         CraftBaseModel.model_config,  # type: ignore[misc]
-        extra="ignore",
+        extra="allow",
```

Comment:
Wouldn't extra fields specified in the craft spread file be ignored when generating the real spread.yaml? For us to keep control on what gets into the final file it could be better to add to the schema, or error if an undesired field is present.

Reply:
Part of the reason for this change was that snapcraft's spread.yaml includes things that aren't in our model, so the pre-processed version with "ignore" breaks.

Since spread works approximately the same as pydantic's "allow," I think it's appropriate to replicate that.

Reply:
I think we can allow extras inside backend definitions, as long as they're passed along, but keeping control at what we allow in toplevel can be safer for now. In any case, we can't make additional toplevel definitions disappear when generating the full spread.yaml file.

Reply:
Using "ignore" for the root model would make additional top-level definitions disappear, "allow" keeps them around.